odoo/orm/models.py

```diff
@@ -4231,12 +4233,33 @@ def create(self, vals_list: list[ValuesType]) -> Self:
         :raise UserError: if a loop would be created in a hierarchy of objects a result of the operation
           (such as setting an object as its own parent)
         """
+        if not isinstance(vals_list, list):
+            # even if the signature indicates list, sometimes iterators are
+            # passed, in that case, make it work as a list
```

Comment:
```suggestion
            # even if the signature indicates list, sometimes iterables are
            # passed, in that case, make it work as a list
```

Reply:
No, sometimes *iterators* are passed. Other iterables like collections would work fine.

Reply:
I agree, but technically speaking, in Python all _iterators_ are _iterables_.  This is awful, since iterators can only be iterated once.  But that's reality.
I suggest to modify `create`'s type signature accordingly.